code/game/objects/explosion.dm

```diff
@@ -131,13 +132,19 @@
 		else
 			dist = 0
 
-		//------- TURF FIRES -------
+		//------- EX_ACT AND TURF FIRES -------
 
 		if(T)
 			if(flame_dist && prob(40) && !isspaceturf(T) && !T.density)
 				PoolOrNew(/obj/effect/hotspot, T) //Mostly for ambience!
 			if(dist > 0)
-				T.ex_act(dist)
+				var/t_type = T.type
+				var/turf/NT = T.ex_act(dist)
+				if(!istype(NT))
+					WARNING("Turf ex_act for type [t_type] did not return the new turf")
```

Comment:
Turfs are static references.

if I store a reference to a floor turf, and later on, that turf becomes a space, that reference now holds the space turf. This is automatic in byond because with the exception of temp turf (turfs created without a loc), turf references are just their x/y/z and when accessing a turf it resolves to a lookup in the game world's 3dim array of turfs.